tests/IAM/Domain/User/Model/PasswordHashTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Twitter\Tests\IAM\Domain\User\Model;
+
+use InvalidArgumentException;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\Attributes\Group;
+use PHPUnit\Framework\Attributes\Test;
+use PHPUnit\Framework\TestCase;
+use Twitter\IAM\Domain\User\Model\Exception\InvalidPasswordException;
+use Twitter\IAM\Domain\User\Model\PasswordHash;
+
+#[Group('unit')]
+#[CoversClass(PasswordHash::class)]
+class PasswordHashTest extends TestCase
+{
+    private const string VALID_PASSWORD = 'Pswd.123';
+
+    #[Test]
+    public function hashesAndVerifiesAValidPlainPassword(): void
+    {
+        $passwordHash = PasswordHash::fromPlainPassword(self::VALID_PASSWORD);
+
+        self::assertTrue(password_verify(self::VALID_PASSWORD, (string) $passwordHash));
+    }
+
+    #[Test]
+    public function rejectsATooShortPlainPassword(): void
+    {
+        self::expectException(InvalidPasswordException::class);
+
+        $shortPassword = substr(self::VALID_PASSWORD, 0, -1);
+        PasswordHash::fromPlainPassword($shortPassword);
+    }
+
+    #[Test]
+    public function createsAValueObjectFromExistingHash(): void
+    {
+        $passwordHash = PasswordHash::fromPlainPassword(self::VALID_PASSWORD);
+
+        self::assertTrue(
+            password_verify(
+                self::VALID_PASSWORD,
+                (string) $passwordHash,
+            ),
+        );
+        self::assertEquals(
+            (string) $passwordHash,
+            (string) PasswordHash::fromHash((string) $passwordHash),
+        );
+    }
+
+    #[Test]
+    public function rejectsAnInvalidPasswordHash(): void
+    {
+        self::expectException(InvalidArgumentException::class);
+
+        PasswordHash::fromHash('abracadabra');
+    }
+
+    #[Test]
+    #[DataProvider('passwordVerificationProvider')]
+    public function verifiesPlainPasswordsCorrectly(string $input, bool $expected): void
+    {
+        $passwordHash = PasswordHash::fromPlainPassword(self::VALID_PASSWORD);
+
+        self::assertSame($expected, $passwordHash->verify($input));
+    }
+
+    public static function passwordVerificationProvider(): array
```

Comment:
@adnlv There is a performance optimization is to use `Generator` as the return type for Data Provider.

Same thing with invalid values.

```suggestion
    public static function passwordVerificationProvider(): Generator
```